athena/solver.py

```diff
@@ -335,6 +335,88 @@ def inference(self, dataset, rank_size=1):
         logging.info("model computation elapsed: %s\ttotal seconds: %s\tRTF: %.4f"
                      % (total_elapsed, total_seconds, float(total_elapsed / total_seconds)))
 
+class GenderSolver(BaseSolver):
```

Comment:
The only difference between GenderSolver and SpeakerClassificationSolver is the number of "top n" we choose. We should probably combine these two solvers into one "ClassificationSolver" and pass top_n as an argument.

Reply:
I think it's okay to merge now. I'll remove these duplicate codes before merging them to the master branch.